src/react/NewGroupPage.js

```diff
@@ -34,202 +34,204 @@ import SingleCardPage from "./ui/SingleCardPage.js";
 import CheckableDeviceTable from "./ui/CheckableDeviceTable.js";
 import { Link } from "react-router-dom";
 
-export default class NewGroupPage extends React.Component {
-  constructor(props) {
-    super(props);
-
-    this.astarte = this.props.astarte;
-
-    this.state = {
-      phase: "loading",
-      isCreatingGroup: false
+export default ({ astarte, history }) => {
+  const [phase, setPhase] = useState("loading");
+  const [groupName, setGroupName] = useState("");
+  const [deviceFilter, setDeviceFilter] = useState("");
+  const [devices, setDevices] = useState([]);
+  const [selectedDevices, setSelectedDevices] = useState(new Set());
+  const [isCreatingGroup, setIsCreatingGroup] = useState(false);
+  const [alertStatus, setAlertStatus] = useState({
+    alertId: 0,
+    alerts: new Map()
+  });
+
+  useEffect(() => {
+    const handleDevicesRequest = (response) => {
+      const deviceList = response.data.map((value) => Device.fromObject(value));
```

Comment:
True, and it will be done there, but sometimes the API returns metadata too.
We will need to build objects that represent the response, a simple array won't cut it, but that's a topic for an other time